doc/user/methods/random_forest.md

```diff
@@ -705,10 +716,107 @@ class CustomCategoricalSplit
+   - `IdentityBootstrap`: no bootstrapping.  Simply copies the input `dataset`, `labels`, and `weights` for each tree's data.
+   - `SequentialBootstrap`: bootstrapping from overlapping sequences such that samples with informational overlap behave more I.I.D.
+     * Useful when data consists of multiple overlapping events (or individual sequences).
+     * `b = SequentialBootstrap(intervals)` will create a `SequentialBootstrap` object, where:
+       - `intervals` is of type `arma::umat`, with 2 rows and `n` columns, where `n` is the number of events to be sampled from.
+       - Each column in `intervals` represents the start and end columns (inclusive) of each event.
+       - So, e.g., if the 10th event is 5 points long, starting at index 6, then column `9` of `intervals` should be `[6, 10]`.
+     * A `SequentialBootstrap` must be passed as the `bootstrap` option to the [advanced constructor](#fully-custom-behavior).
+     * For more information, see: M. López de Prado (2018): "Advances in Financial Machine Learning", pp. 63-65.
+
+ * When using a `BootstrapType` that requires an instantiated object (such as `SequentialBootstrap`), the following advanced constructor forms can be used for `RandomForest`:
+ 
+   - `RandomForest(... , bootstrap)`
+   - `RandomForest(..., bootstrap)`
+
+ * A custom `BootstrapType` class must take a `bool` template parameter `UseWeights` and implement one function:
+
+```c++
+class CustomBootstrapType
+{
+ public:
+  /**
+   * Compute a bootstrap dataset based on the original dataset.
+   * If `UseWeights` is `false`, then `weights` and `bootstrapWeights` can be
+   * ignored.
+   *
+   * When the function is complete, `bootstrapDataset` and `bootstrapLabels`
+   * should contain a bootstrapped dataset.  If `UseWeights` is `true`, then
+   * `bootstrapWeights` should contain the corresponding instance weights for
+   * the bootstrapped dataset.
+   */
+  template<
+      bool UseWeights,
+      typename MatType,
+      typename LabelsType,
+      typename WeightsType>
+  void Bootstrap(
+      const MatType& dataset,
+      const LabelsType& labels,
+      const WeightsType& weights,
+      MatType& bootstrapDataset,
+      LabelsType& bootstrapLabels,
+      WeightsType& bootstrapWeights);
+};
+```
+
+---
+
+Train a `RandomForest` with the `SequentialBootstrap` strategy.
+
+```c++
+// 1000 random points in 10 dimensions. In reality this might be
+// financial time-series data.
+arma::mat dataset(10 /* rows */, 1000 /* cols */, arma::fill::randu);
+
+// Random labels for each point, totaling 5 classes.
+arma::Row<size_t> labels =
+    arma::randi<arma::Row<size_t>>(1000, arma::fill::zeros);
+
+arma::umat intervals(2, labels.n_cols);
+
+// This example only has 2 active events
+intervals(0, 0) = 0; // start of first event
+intervals(1, 0) = 200; // end of first event
+intervals(0, 200) = 200; // start of second event
+intervals(1, 200) = 500; // end of second event
```

Comment:
It feels like this is a confusing example: why have 1000 data points if only 2 of them are active events?  I think it might be more effective to just have a dataset with 3 points or something, and where all of the events are active.  Did I overlook something?  It's important to get this example right because this is what a user will most likely look at when trying to implement their own code, so we probably should also make sure this handles the intended use case, where events are overlapping (they aren't in the example here).

Reply:
only having 3 points and then using a RF with 20 trees also sounds strange. I just added a third point and explained in the comment how they overlap / not overlap